llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -2251,9 +2257,9 @@ static Instruction *foldICmpShlOne(ICmpInst &Cmp, Instruction *Shl,
         Pred = ICmpInst::ICMP_UGT;
     }
 
-    unsigned CLog2 = C.logBase2();
+    unsigned CLog2 = Div.logBase2();
```

Comment:
Can you add a test where `C` and `C2` are not a power of two and we need to roundup the log? I.e
`(48 << X) u>= 144`?

I don't see the ceil logic for that case.

Reply:
https://alive2.llvm.org/ce/z/XejB_d